.github/actions/override-workflow-variables/action.yml

```diff
@@ -0,0 +1,68 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'Override Workflow Variables'
+description: |-
+  Use this action to load/override variables into GITHUB_ENV.
+
+inputs:
+  working_directory:
+    description: 'The working directory to look for the env file.'
+    required: false
+    default: '.'
+  env_file:
+    description: 'The env file containing variables to override with.'
+    required: false
+    default: '.env'
+  require_env:
+    description: 'Require the use of an env file for overriding variables.'
+    required: false
+    default: 'false'
+  enforced_directories:
+    description: 'A CSV string of directories to require the use of an env file for overriding variables.'
+    required: false
+    default: ''
+
+runs:
+  using: 'composite'
+  steps:
+    - name: 'Load Env File'
+      shell: 'bash'
+      working-directory: '${{ inputs.working_directory }}'
+      env:
+        ENV_FILE: '${{ inputs.env_file }}'
+        REQUIRE_ENV: '${{ inputs.require_env }}'
+        ENFORCED_DIRECTORIES: '${{ inputs.enforced_directories }}'
+      run: |-
+        if [ -f "$ENV_FILE" ]; then
+          # Filter out comment lines, empty lines, and lines with only whitespace.
+          grep -v '^[[:space:]]*#\|^$\|^\s*$' $ENV_FILE >> ${GITHUB_ENV}
+        elif [ "$REQUIRE_ENV" == true ]; then
+
+          if [ "$ENFORCED_DIRECTORIES" == "" ]; then
+            # Set to top-level dir of respository to enforce on all subdirectories.
```

Comment:
Maybe make it clear that "all subdirectories" isn't recursive, it's just immediate children

Reply:
I changed the wording from subdirectories to working-directories.

e.g. If the `enforced_directories = 'foo/'`, then a working directory of `foo/bar/baz` and `foo/bar` will require `foo/bar/baz/.env` and `foo/bar/.env` respectively.

Does this make it clearer?